internal/pkg/remote/client.go

```diff
@@ -220,13 +220,30 @@ func (c *Client) Send(
 			c.log.With("error", err).Debugf(msg)
 			continue
 		}
+		c.checkApiVersionHeaders(reqID, resp)
 
 		return resp, nil
 	}
 
 	return nil, fmt.Errorf("all hosts failed: %w", multiErr)
 }
 
+func (c *Client) checkApiVersionHeaders(reqID string, resp *http.Response) {
+	const elasticApiVersionHeaderKey = "Elastic-Api-Version"
```

Comment:
Could you use the package-level `elasticApiVersionHeaderKey` const instead of redefining it in local scope here?

Reply:
Ah yes, my bad.  I thought this file here was in the `client` package. Never mind :)